### PR TITLE
chore: remove more unused code

### PR DESCRIPTION
**Pull request by incendial**
After the mention [here](https://github.com/flutter/devtools/pull/5918#discussion_r1232529612) I ran the unused check again with a different flag and got a bunch of new highlights. Some of them look illegal 😅.  

Note: not all of the removed code might be unused intentionally, but I have no context of that. Also, there are might be not obvious to me false-positives.

RELEASE_NOTE_EXCEPTION=[Internal refactoring]

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [-] I listed at least one issue that this PR fixes in the description above.
- [-] I updated/added relevant documentation (doc comments with `///`).
- [-] I added new tests to check the change I am making, or there is a reason for not adding tests.

### CODE REVIEW

**Comment by incendial** [comment]
@jacob314 @polina-c fyi

**Comment by incendial** [comment]
Hm, I didn't notice that `devtools_shared` is a package you publish to pub, so I expect some of them to be reverted, but waiting for the review.

**Comment by polina-c** [comment]
Looks good. Some questions:

1. Can you please test that Memory > Chart is still functioning?
2. What is our process around removing unused code? Is there any continuous automation that kicks it off?

**Comment by polina-c** [comment]
@elliette @kenzieschmoll @CoderDake 

Can you also review?

**Comment by incendial** [comment]
> What is our process around removing unused code?

You mean you want to run `dcm unused code` checks on bots?

**Comment by incendial** [comment]
Here is the video of working app:

https://github.com/flutter/devtools/assets/21116742/5c61f472-b8f4-402c-a5fb-ec583763843e

Looks like everything works except for one overflow error (but I'm not sure whether it existed before):

https://github.com/flutter/devtools/assets/21116742/1a36695e-8e18-4345-93ed-aa0543d804b6

**Review by polina-c — APPROVED** [review]

**Comment by polina-c** [comment]
LGTM assuming others LGTM their parts

File: packages/devtools_app/lib/src/shared/eval_on_dart_library.dart
Comment (CoderDake) [inline packages/devtools_app/lib/src/shared/eval_on_dart_library.dart:651]: probably better to add scope to toString

File: packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart
Comment (CoderDake) [inline packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart:72]: it might be better to add this to toString

**Review by CoderDake — COMMENTED** [review]
I have some concerns about devtools_shared and some other small comments

**Review by kenzieschmoll — COMMENTED** [review]
please remove all changes to `packages/devtools_shared`. The `devtools_shared` package is published and is used by the devtools server, which is part of the Dart SDK.

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_context_menu.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_context_menu.dart:25]: @polina-c this seems like a bug that this is unused?

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/memory_timeline.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/shared/primitives/memory_timeline.dart:120]: @polina-c this also seems like a bug that we have stopped adding an event to the timeline when we take a snapshot. Maybe the action item here is to use this code.

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart:362]: @bkonyi is this a bug that this code is unused?

File: packages/devtools_app/lib/src/service/service_extensions.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/service/service_extensions.dart:60]: can you revert changes to this file? we have an outstanding issue to add the UI control for the `togglePlatformMode` service extension, at which point this would need to be added back. https://github.com/flutter/devtools/issues/2780

File: packages/devtools_app/lib/src/shared/diagnostics/diagnostics_node.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/diagnostics_node.dart:11]: I recommend that we remove the changes to this file. This class is extended and used by a customer inside google, so we do not want to risk breaking them. Also, we some of these helpers will be useful as we continue to evolve the Inspector and Layout Explorer and add advanced functionality.

File: packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart:688]: this method is used by an internal customer. Please revert.

File: packages/devtools_app/lib/src/shared/memory/adapted_heap_data.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/memory/adapted_heap_data.dart:67]: it looks like the graph parameter is being used here, so is this removal a false positive?

**Comment by Hixie** [comment]
@incendial Thanks for your contribution! Is this still on your radar?

**Comment by incendial** [comment]
@Hixie hi, thanks for the reminder, I hope to get back to this PR soon.

Btw, any chance we can discuss whether the Flutter team could benefit from using Dart Code Metrics?

**Comment by Hixie** [comment]
@incendial There's been some discussion of that, I forget the conclusion. I recommend bringing it up on the #hackers-framework channel of our [Discord](https://github.com/flutter/flutter/wiki/Chat).

File: packages/devtools_app/lib/src/shared/eval_on_dart_library.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/eval_on_dart_library.dart]: Done

File: packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart:72]: Sure, do you have any preference regarding the format (ex. '$file:$line:$column' or '$file\n$line:$column')?

File: packages/devtools_app/lib/src/shared/eval_on_dart_library.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/eval_on_dart_library.dart]: Done for all 3, although not sure whether the scope should go last. Can you please take a look?

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_context_menu.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_context_menu.dart:25]: Should I add a todo or can you suggest what changes are expected to fix it?

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/memory_timeline.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/memory/shared/primitives/memory_timeline.dart:120]: Same as https://github.com/flutter/devtools/pull/5949#discussion_r1342165085

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart:362]: Same as https://github.com/flutter/devtools/pull/5949#discussion_r1342165085

File: packages/devtools_app/lib/src/service/service_extensions.dart
Comment (incendial) [reply packages/devtools_app/lib/src/service/service_extensions.dart]: Done

File: packages/devtools_app/lib/src/shared/diagnostics/diagnostics_node.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/diagnostics/diagnostics_node.dart:11]: Sure, done.

File: packages/devtools_app/lib/src/shared/memory/adapted_heap_data.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/memory/adapted_heap_data.dart:67]: Only the isolateId was removed here (from 2 parameters `graph and `isolateId` to only `graph` remaining).

File: packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/diagnostics/inspector_service.dart]: Done

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_context_menu.dart
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_context_menu.dart:25]: @polina-c please advise

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/memory_timeline.dart
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/memory/shared/primitives/memory_timeline.dart:120]: @polina-c please advise

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart:362]: @bkonyi please advise

**Review by kenzieschmoll — COMMENTED** [review]
please add a changelog entry for package:devtools_app_shared, which is also published on pub. This package is still under early development so these breaking changes are okay. Please put the changes under a new entry for version 0.0.6 and bump the package version in the pubspec.yaml to 0.0.6

File: packages/devtools_app/lib/src/shared/primitives/utils.dart
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/shared/primitives/utils.dart:455]: please add this extension method back as this is useful for debugging purposes

File: packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart:72]: '$file:$line:$column'  SGTM

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart:362]: Waiting for this one and then will address all latest comments and bump the version.

File: packages/devtools_app/lib/src/shared/eval_on_dart_library.dart
Comment (CoderDake) [reply packages/devtools_app/lib/src/shared/eval_on_dart_library.dart]: LGTM :D

File: packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart
Comment (bkonyi) [reply packages/devtools_app/lib/src/screens/profiler/cpu_profile_model.dart:362]: Sorry for the delay on this.

I think it can be removed, it was probably added for symmetry and then never used.

File: packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart
Comment (incendial) [reply packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart:72]: What bothers me is that file is not passed anywhere and is not used in `bool operator ==(other) {` and `int get hashCode =>`. On top of that any `toString` call will start with `null:`.

Are you sure this field should be restored?

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_context_menu.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_context_menu.dart:25]: I see that this has already been addressed, I'm not adding it then

File: packages/devtools_app/lib/src/screens/memory/shared/primitives/memory_timeline.dart
Comment (incendial) [reply packages/devtools_app/lib/src/screens/memory/shared/primitives/memory_timeline.dart:120]: This one as well

**Comment by incendial** [comment]
@kenzieschmoll I've updated the package version to 0.0.7 and updated the changelog. Can you please take a look?
The only unresolved conversation is this one https://github.com/flutter/devtools/pull/5949#discussion_r1367683654. Are you sure you want this field back?

And as for failing pipelines, looks like they are failing not because of the changes introduced here. The commit before I upstreamed the main branch got all checks passed.

File: packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart:72]: @elliette does this seem like a bug that we are not using the `file` parameter anywhere? I suspect this would cause weird equality issues where two `SourcePosition` objects appear identical because they have the same `line` & `column` & `tokenPos` but they are actually from two different files.

**Comment by kenzieschmoll** [comment]
> And as for failing pipelines, looks like they are failing not because of the changes introduced here. The commit before I upstreamed the main branch got all checks passed.

If you merge master, these issues should be resolved.

File: packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart
Comment (elliette) [reply packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart:72]: I don't think this is a bug. It looks like `file` parameter was added in PR https://github.com/flutter/devtools/pull/3448 to combine two definitions of `SourcePosition` in two different files (`debugger_model` and `diagnostics_node`), the second of which had `file` as a parameter but never used it. 

`SourcePosition` is usually used in conjunction with some other class that contains file information (for example `ScriptRef` in [`StackFrameAndSourcePosition`](https://github.com/flutter/devtools/blob/6a2271a81ecea41aeb77ffed617587f1cceccc3d/packages/devtools_app/lib/src/screens/debugger/debugger_model.dart#L189)).

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/shared/diagnostics/primitives/source_location.dart:72]: Okay thanks for the input - @incendial why don't we just remove this parameter completely then as you originally did.

**Comment by incendial** [comment]
@kenzieschmoll I think all threads are now address + the pipelines are green. Can you please take a final look? Hope we can merge this today 😁

**Pull request merged**
